### PR TITLE
Add MosaicML inference endpoints

### PR DESCRIPTION
**Pull request by dakinggg**
# Add MosaicML inference endpoints
This PR adds support in langchain for MosaicML inference endpoints. We both serve a select few open source models, and allow customers to deploy their own models using our inference service. Docs are here (https://docs.mosaicml.com/en/latest/inference.html), and sign up form is here (https://forms.mosaicml.com/demo?utm_source=langchain). I'm not intimately familiar with the details of langchain, or the contribution process, so please let me know if there is anything that needs fixing or this is the wrong way to submit a new integration, thanks!

I'm also not sure what the procedure is for integration tests. I have tested locally with my api key.

## Who can review?
@hwchase17

### CODE REVIEW

File: langchain/embeddings/mosaicml.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+                "Could not find MOSAICML_API_TOKEN in environment or as a parameter."
+            )
+        values["mosaicml_api_token"] = mosaicml_api_token
+        return values
+
+    @property
+    def _identifying_params(self) -> Mapping[str, Any]:
+        """Get the identifying parameters."""
+        return {
+            **{"endpoint_url": self.endpoint_url},
```
Comment (dev2049) [inline langchain/embeddings/mosaicml.py:66]: could just do
```python
return {"endpoint_url": self.endpoint_url}
```

File: langchain/embeddings/mosaicml.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    )
+    """Endpoint URL to use."""
+    embed_instruction: str = "Represent the document for retrieval: "
+    """Instruction used to embed documents."""
+    query_instruction: str = (
+        "Represent the question for retrieving supporting documents: "
+    )
+    """Instruction used to embed the query."""
+
+    mosaicml_api_token: Optional[str] = None
```
Comment (dev2049) [inline langchain/embeddings/mosaicml.py:42]: could probably just type as
```python
mosaicml_api_token: str
```

File: langchain/llms/mosaicml.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        _model_kwargs = self.model_kwargs or {}
+        return {
+            **{"endpoint_url": self.endpoint_url},
+            **{"model_kwargs": _model_kwargs},
+        }
+
+    @property
+    def _llm_type(self) -> str:
+        """Return type of llm."""
+        return "mosaicml_llm"
```
Comment (dev2049) [inline langchain/llms/mosaicml.py:91]: don't think _llm suffix is needed

**Review by dev2049 — COMMENTED** [review]
couple comments but mostly looks good!

File: langchain/llms/mosaicml.py (hunk 3 above)
Comment (dakinggg) [reply langchain/llms/mosaicml.py:91]: sure, I wasn't clear where this was used, will change

File: langchain/embeddings/mosaicml.py (hunk 2 above)
Comment (dakinggg) [reply langchain/embeddings/mosaicml.py:42]: pydantic complains if i do this. I'm not familiar with pydantic and the whole initialization setup here so I'll just leave as optional unless you're opposed.

**Pull request merged**